### ISSUE TITLE
ST003 & ST004: ExtractVariables null check fails for dotted variable names

### ISSUE THREAD

**Issue by SeboRei** [apigee/apigeelint#594]
## Summary
The ExtractVariables payload check plugins (ST003, ST004) incorrectly flag valid null checks as missing when the Source variable contains dots in its name (e.g., `private.tokenexchangeResponse`).

## Environment
- apigeelint version: latest
- Node.js version: 20.x
- OS: Windows/WSL Ubuntu 24.04

## Affected Files
- `lib/package/plugins/_extractVariablesAttachmentCheck.js`
- `lib/package/plugins/_extractVariablesPayloadCheck.js`

## Steps to Reproduce

1. Create an ExtractVariables policy with a custom Source containing dots:
```xml
<ExtractVariables name="M-ExVa-ExtractExchangedToken">
  <Source>private.tokenexchangeResponse</Source>
  <JSONPayload>
    <Variable name="access_token">
      <JSONPath>$.access_token</JSONPath>
    </Variable>
  </JSONPayload>
</ExtractVariables>
```

2. Add a proper null check in the Step condition:
```xml
<Step>
  <Name>M-ExVa-ExtractExchangedToken</Name>
  <Condition>
    private.tokenexchangeResponse.content != null
  </Condition>
</Step>
```

3. Run apigeelint on the bundle

## Expected Behavior
The linter should recognize `(private.tokenexchangeResponse.content != null)` as a valid null check and not report any warnings.

## Actual Behavior
The linter reports:
```
warning: For the ExtractVariables step (M-ExVa-ExtractExchangedToken), an appropriate check for a message body was not found. 
For example, use <Condition>private.tokenexchangeResponse.content != null</Condition>
```

## Root Cause

The issue is in the `stringNonNullCheckRegexp` function which uses word boundaries (`\b`) around the source variable name:

```javascript
const stringNonNullCheckRegexp = (source) => 
  `\\(? *\\b${source}\\b *(!=|NotEquals|IsNot) *(\"\"|null) *\\)?`;
```

### Problems:

1. **Word boundaries don't work with dots**: The `\b` regex word boundary only matches between word characters (`[a-zA-Z0-9_]`) and non-word characters. Since `.` is a non-word character, `\bprivate.tokenexchangeResponse\b` fails to match the variable name correctly.

2. **Dots not escaped**: The dots in the source variable are inserted into the regex without escaping, causing them to act as wildcards instead of literal dots.

3. **Newlines not matched**: The pattern uses `.*` which doesn't match newlines by default in JavaScript. Multi-line conditions fail to match.

## Proposed Fix

```javascript
const stringNonNullCheckRegexp = (source) => {
  // Escape dots for regex literal matching
  const escapedSource = source.replaceAll('.', String.raw`\.`);
  // Don't use word boundaries for dotted names - they break the match
  const sourcePattern = source.includes('.') ? escapedSource : String.raw`\b${source}\b`;
  return String.raw`\(? *${sourcePattern} *(!=|NotEquals|IsNot) *(""|null) *\)?`;
};
```

Additionally, change the condition regex wrapper to support newlines:
```javascript
// Before:
condRegExp = `^.*${condRegExp}.*$`;

// After:
condRegExp = String.raw`^[\s\S]*` + condRegExp + String.raw`[\s\S]*$`;
```

And replace `.match()` with `new RegExp().exec()` for better performance:
```javascript
// Before:
if (condition && condition.getExpression().trim().match(condRegExp)) {

// After:
if (condition && new RegExp(condRegExp).exec(condition.getExpression().trim())) {
```

## Impact
This bug affects any Apigee proxy or shared flow that uses:
- Custom message variables with dots in the name (e.g., `private.variableName`, `flow.customVar`)
- Multi-line conditions
- ExtractVariables policies with JSONPayload or XMLPayload

## Testing
After applying the fix, the following test cases should pass:

1. Simple variable names (backward compatibility): `request`, `response`, `message`
2. Dotted variable names: `private.tokenexchangeResponse`, `flow.customVar`
3. Multi-line conditions with newlines
4. Deeply nested variable names: `org.project.module.variable`

**Comment by SeboRei** [comment]
Verified from my side, apparently was fixed between my forked version and upstream.
Sorry for the inconvenience.
